components/home/lists/lists-controller.js

```diff
@@ -436,6 +436,18 @@ trackerCapture.controller('ListsController',function(
         $scope.canSyncLabTests = true;
         $scope.syncLabTests = function () {
             $scope.canSyncLabTests = false;
+            if($scope.useLabTestForProgram($scope.selectedProgram)) {
+                var userId;
+                try{
+                    userId = JSON.parse(sessionStorage.USER_PROFILE).id
+                }
+                finally {}
+                FNrLookupService.startLabTestSync($scope.selectedOrgUnit.code, userId).then(function(svar){
+                    if(svar) {
+                        $scope.canSyncLabTests = false;
```

Comment:
I would think the button should be disabled right away, not wait for the promise?

Reply:
I see that it is allready disabled on line 438. But I should add logic for enable it if the call fails, rather than disable it if it succeeds. Will fix